vmware_dvs/utils/dvs_util.py

```diff
@@ -512,13 +512,14 @@ def create_port_map(dvs_list):
     return port_map
 
 
-def get_dvs_by_id_and_key(dvs_list, port_id, port_key):
+def get_dvs_by_network(dvs_list, network_id):
     for dvs in dvs_list:
-        port = dvs._get_port_info_by_portkey(port_key)
-        if port:
-            if port.config.name == port_id:
+        try:
+            network_name = dvs._get_net_name(dvs.dvs_name, {'id': network_id})
+            if dvs._get_pg_by_name(network_name):
```

Comment:
You does not need PG object itself. In multiple VM creation cases it cause many identical calls to vCenter. I hope that you can improve it by cashing of network names for each DVS and try to find dvs in cash at first.